### PR TITLE
Community PR

### PR DESCRIPTION
**Pull request by mviswanathsai**
Added community controllers, routes and models

### CODE REVIEW

File: src/controllers/communityController.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                res.status(400).json({
+                    code: 400,
+                    message: "community not found",
+                });
+            }
+        }
+    );
+};
+
+export const getAllTripsOfCommunity = async (req, res) => {
```
Comment (AdityaRaimec22) [inline src/controllers/communityController.js]: You can remove this function as I created it in my PR where I applied the functionality of applying different filters like user can check put the filter of origin, destination, date and time range it is a fully tested function which works as expected.
While this function only gives the trips and user can't query trips

**Review by AdityaRaimec22 — CHANGES_REQUESTED** [review]

File: src/controllers/communityController.js (hunk 1 above)
Comment (mviswanathsai) [reply src/controllers/communityController.js]: That's great! the aim of this PR was just to give comprehensive overview of the architecture. I will make the changes as necessary.

File: src/routes/communityRouter.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+const router = express.Router();
+
+
+router.route("/").get(getAllCommunities).post(createCommunity)
+router.route("/:community_id").get(getCommunityById).put(updateCommunity).delete(deleteCommunity)
+router.route("/:community_id/trips").get(getAllTripsOfCommunity)
+router.route("/:community_id/trips/:trip_id").get().post(addTripToCommunity).delete(removeTripFromCommunity)
+
+// TODO: add the function that corresponds to getTripById here at the get endpoint of /:trip_id
+export default router;
```
Comment (AdityaRaimec22) [inline src/routes/communityRouter.js]: This router is exported but not imported anywhere. Means this file is just created and not used anywhere.

Comment (AdityaRaimec22) [inline src/routes/communityRouter.js:13]: This router is exported but no where imported.

**Review by Varun-Kolanu — DISMISSED** [review]

File: src/routes/communityRouter.js (hunk 2 above)
Comment (mviswanathsai) [reply src/routes/communityRouter.js]: const communityRouter = require("./src/routes/communityRouter.js")

**Pull request merged**